### PR TITLE
Configurable Maximum Decimal Digits Number

### PR DESCRIPTION
**Pull request by efguydan**
This Pull Request makes the maximum amount of decimal digits number configurable. It exposes a method and an XML attribute to configure it. Tests are also included to ensure functionality.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,6 +67,20 @@ The `CurrencyEditText` uses the default `Locale` if no locale is specified. `Loc
...
  ```
+ 
+### Decimal Places
+The maximum number of decimal digits can be specified using the `maxNumberOfDecimalDigits` attributes in the xml, requiring
+a minimum value of 1. It has a default value of 2.
+
+```xml
+   <com.cottacush.android.currencyedittext.CurrencyEditText
+            ...
+            app:maxNumberOfDecimalDigits="3" />
```
Comment (rasheedsulayman) [inline README.md:78]: I think we should still default to 2. 2 is the most common.

File: sample/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -11,6 +11,7 @@
             app:currencySymbol="$"
             app:useCurrencySymbolAsHint="true"
             app:localeTag="en-NG"
+            app:maxNumberOfDecimalDigits="3"
```
Comment (rasheedsulayman) [inline sample/src/main/res/layout/activity_main.xml]: I think we should still default to 2 in the sample app. 2 is the most common.

File: library/src/main/java/com/cottacush/android/currencyedittext/utils/Constants.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cottacush.android.currencyedittext.utils
+
+object Constants {
+
+    object Messages {
+        const val MAX_DECIMAL_DIGITS_ERROR = "Maximum number of Decimal Places must be a positive integer"
```
Comment (rasheedsulayman) [inline library/src/main/java/com/cottacush/android/currencyedittext/utils/Constants.kt]: I don't think we need to extract the error message to a constant. It's usually more readable to leave an exception message directly with the throw statement.

**Comment by rasheedsulayman** [comment]
Did you maybe check its behavior for really long decimal places like 6 or 7?
Also some weird Locales with uncommon comma & decimal separators. Just trying to make sure we don't break anything.

**Comment by efguydan** [comment]
> Did you maybe check its behavior for really long decimal places like 6 or 7?
> Also some weird Locales with uncommon comma & decimal separators. Just trying to make sure we don't break anything.

Yeah, there are tests for 7 and 10 decimal digits. Also, there are tests for 3 different locales accomodating different grouping and decimal separators. All existing and new tests are passing, so I think nothing should be broken.

**Pull request merged**